File: web/frontend/src/api/configs.ts
import api from './client'

export const getTemplates = (params?: Record<string, any>) => api.get('/configs/templates', { params })
export const getTemplate = (id: number) => api.get(`/configs/templates/${id}`)
export const createTemplate = (data: Record<string, any>) => api.post('/configs/templates', data)
export const updateTemplate = (id: number, data: Record<string, any>) => api.put(`/configs/templates/${id}`, data)
export const deleteTemplate = (id: number) => api.delete(`/configs/templates/${id}`)
export const getVersions = (templateId: number) => api.get(`/configs/templates/${templateId}/versions`)
export const createVersion = (templateId: number, data: Record<string, any>) => api.post(`/configs/templates/${templateId}/versions`, data)
export const getVersion = (versionId: number) => api.get(`/configs/versions/${versionId}`)

export const getModules = (params?: Record<string, any>) => api.get('/configs/modules', { params })
export const getModule = (id: number) => api.get(`/configs/modules/${id}`)
export const createModule = (data: Record<string, any>) => api.post('/configs/modules', data)
export const updateModule = (id: number, data: Record<string, any>) => api.put(`/configs/modules/${id}`, data)
export const deleteModule = (id: number) => api.delete(`/configs/modules/${id}`)
export const getModuleVersions = (moduleId: number) => api.get(`/configs/modules/${moduleId}/versions`)
export const createModuleVersion = (moduleId: number, data: Record<string, any>) => api.post(`/configs/modules/${moduleId}/versions`, data)

export const previewRenderedConfig = (data: Record<string, any>) => api.post('/configs/rendered-configs/preview', data)
export const getRenderedConfig = (id: number) => api.get(`/configs/rendered-configs/${id}`)

File: internal/routers/configs.go
package routers

import (
	"github.com/fluent-manager/fluent-manager/internal/middleware"
	"github.com/gin-gonic/gin"
)

func registerConfigRoutes(authed *gin.RouterGroup, h *allHandlers) {
	// Config Templates & Modules
	configs := authed.Group("/configs")
	{
		configs.GET("/templates", middleware.RequirePermission("configs", "read"), h.Config.ListTemplates)
		configs.GET("/templates/:id", middleware.RequirePermission("configs", "read"), h.Config.GetTemplate)
		configs.POST("/templates", middleware.RequirePermission("configs", "create"), h.Config.CreateTemplate)
		configs.PUT("/templates/:id", middleware.RequirePermission("configs", "update"), h.Config.UpdateTemplate)
		configs.DELETE("/templates/:id", middleware.RequirePermission("configs", "delete"), h.Config.DeleteTemplate)
		configs.GET("/templates/:id/versions", middleware.RequirePermission("configs", "read"), h.Config.ListVersions)
		configs.POST("/templates/:id/versions", middleware.RequirePermission("configs", "create"), h.Config.CreateVersion)
		configs.GET("/versions/:version_id", middleware.RequirePermission("configs", "read"), h.Config.GetVersion)
		configs.GET("/modules", middleware.RequirePermission("configs", "read"), h.Config.ListModules)
		configs.GET("/modules/:id", middleware.RequirePermission("configs", "read"), h.Config.GetModule)
		configs.POST("/modules", middleware.RequirePermission("configs", "create"), h.Config.CreateModule)
		configs.POST("/modules/batch-delete", middleware.RequirePermission("configs", "delete"), h.Config.DeleteModules)
		configs.PUT("/modules/:id", middleware.RequirePermission("configs", "update"), h.Config.UpdateModule)
		configs.DELETE("/modules/:id", middleware.RequirePermission("configs", "delete"), h.Config.DeleteModule)
		configs.GET("/modules/:id/versions", middleware.RequirePermission("configs", "read"), h.Config.ListModuleVersions)
		configs.POST("/modules/:id/versions", middleware.RequirePermission("configs", "create"), h.Config.CreateModuleVersion)
		configs.POST("/rendered-configs/preview", middleware.RequirePermission("configs", "read"), h.Config.PreviewRenderedConfig)
		configs.GET("/rendered-configs/:id", middleware.RequirePermission("configs", "read"), h.Config.GetRenderedConfig)
		configs.GET("/pipelines", middleware.RequirePermission("configs", "read"), h.Config.ListPipelines)
		configs.GET("/pipelines/:id", middleware.RequirePermission("configs", "read"), h.Config.GetPipeline)
		configs.POST("/pipelines", middleware.RequirePermission("configs", "create"), h.Config.CreatePipeline)
		configs.PUT("/pipelines/:id", middleware.RequirePermission("configs", "update"), h.Config.UpdatePipeline)
		configs.DELETE("/pipelines/:id", middleware.RequirePermission("configs", "delete"), h.Config.DeletePipeline)
	}

	// Config Analysis
	analysis := authed.Group("/config-analysis")
	{
		analysis.POST("/log-sample-assistant", middleware.RequirePermission("configs", "read"), h.AI.AnalyzeLogSample)
		analysis.POST("/import-existing", middleware.RequirePermission("configs", "read"), h.FluentOps.ImportExistingConfig)
		analysis.POST("/lint", middleware.RequirePermission("configs", "read"), h.FluentOps.LintConfig)
		analysis.POST("/replay", middleware.RequirePermission("configs", "read"), h.FluentOps.ReplayConfig)
		analysis.POST("/diff", middleware.RequirePermission("configs", "read"), h.FluentOps.SemanticDiff)
		analysis.POST("/compatibility", middleware.RequirePermission("configs", "read"), h.FluentOps.CheckCompatibility)
		analysis.GET("/:id", middleware.RequirePermission("configs", "read"), h.FluentOps.GetAnalysisResult)
	}
}
